### PR TITLE
[search] Refactor Results.

### PR DESCRIPTION
**Pull request by ygorshenin**

### CODE REVIEW

File: search/result.cpp [hunk 1]
```diff
@@ -3,29 +3,55 @@
...
 
+#include <algorithm>
+
+using namespace std;
+
 namespace search
 {
+namespace
+{
+// Following methods joins only non-empty arguments in order with
```
Comment (mpimenov) [inline search/result.cpp]: s/joins/join/

File: search/result.cpp [hunk 2]
```diff
@@ -3,29 +3,55 @@
...
+// comma.
```
Comment (mpimenov) [inline search/result.cpp]: s/comma/commas/ ?

File: search/result.cpp [hunk 3]
```diff
@@ -134,12 +143,31 @@ string Result::ToStringForStats() const
...
+  case Result::Type::PureSuggest: return "PureSuggest";
+  case Result::Type::SuggestFromFeature: return "SuggestFromFeature";
+  }
+
+  return "Unknown";
+}
+
+string DebugPrint(Result const & result)
+{
+  return "Result { Name: " + result.GetString() + "; Type: " + result.GetFeatureTypeName() +
```
Comment (mpimenov) [inline search/result.cpp]: s/{/[/ ?

File: search/result.hpp [hunk 4]
```diff
@@ -6,145 +6,155 @@
...
+    PureSuggest,
+    SuggestFromFeature
   };
 
-  /// Metadata for search results. Considered valid if m_resultType == RESULT_FEATURE.
+  // Metadata for search results. Considered valid if GetResultType() == Type::Feature.
   struct Metadata
   {
-    string m_cuisine;                              // Valid only if not empty. Used for restaurants.
+    // Valid only if not empty, Used for restaurants.
```
Comment (mpimenov) [inline search/result.hpp]: s/,/./

File: search/result.hpp [hunk 5]
```diff
@@ -153,33 +163,34 @@ class Results
 
   bool AddResult(Result && result);
 
-  // Fast version of AddResult() that doesn't do any duplicates checks.
+  // Fast version of AddResult() that doesn't do any duplicates
```
Comment (mpimenov) [inline search/result.hpp]: s/duplicates checks/checks for duplicates/

File: search/result.cpp (hunk 1 above)
Comment (ygorshenin) [reply search/result.cpp]: Done

File: search/result.cpp (hunk 2 above)
Comment (ygorshenin) [reply search/result.cpp]: Done

File: search/result.hpp (hunk 4 above)
Comment (ygorshenin) [reply search/result.hpp]: s/U/u/ ?

File: search/result.hpp (hunk 5 above)
Comment (ygorshenin) [reply search/result.hpp]: Done

File: search/result.hpp (hunk 4 above)
Comment (mpimenov) [reply search/result.hpp]: Also possible

File: search/result.cpp (hunk 3 above)
Comment (ygorshenin) [reply search/result.cpp]: Done

**Comment by ygorshenin** [comment]
@mpimenov PTAL

**Pull request merged**